File: src/main/scala/org/openapitools/client/api/ExportSendigVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportSendigVariablesRow._

case class ExportSendigVariablesRow (
  version: Option[String],
variableOrder: Option[String],
`class`: Option[String],
datasetName: Option[String],
variableName: Option[String],
variableLabel: Option[String],
`type`: Option[String],
controlledTermsCodelistOrFormat: Option[String],
role: Option[String],
cDISCNotes: Option[String],
core: Option[String])

object ExportSendigVariablesRow {
  import DateTimeCodecs._

  implicit val ExportSendigVariablesRowCodecJson: CodecJson[ExportSendigVariablesRow] = CodecJson.derive[ExportSendigVariablesRow]
  implicit val ExportSendigVariablesRowDecoder: EntityDecoder[ExportSendigVariablesRow] = jsonOf[ExportSendigVariablesRow]
  implicit val ExportSendigVariablesRowEncoder: EntityEncoder[ExportSendigVariablesRow] = jsonEncoderOf[ExportSendigVariablesRow]
}

File: src/cdisc_library_api/specs/export_sendig_variables_row.clj
(ns cdisc-library-api.specs.export-sendig-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-sendig-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :VariableOrder) string?
   (ds/opt :Class) string?
   (ds/opt :DatasetName) string?
   (ds/opt :VariableName) string?
   (ds/opt :VariableLabel) string?
   (ds/opt :Type) string?
   (ds/opt :ControlledTermsCodelistorFormat) string?
   (ds/opt :Role) string?
   (ds/opt :CDISCNotes) string?
   (ds/opt :Core) string?
   })

(def export-sendig-variables-row-spec
  (ds/spec
    {:name ::export-sendig-variables-row
     :spec export-sendig-variables-row-data}))
